internal/sidecarlogresults/sidecarlogresults_test.go

```diff
@@ -0,0 +1,71 @@
+package sidecarlogresults
+
+import (
+	"io/fs"
+	"log"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+func TestLookForResults(t *testing.T) {
+	for _, c := range []struct {
+		desc         string
+		resultName   string
+		resultValue  string
+		createResult bool
+		stepError    bool
+	}{{
+		desc:         "good result",
+		resultName:   "foo",
+		resultValue:  "bar",
+		createResult: true,
+		stepError:    false,
+	}, {
+		desc:         "empty result",
+		resultName:   "foo",
+		resultValue:  "",
+		createResult: true,
+		stepError:    true,
+	}, {
+		desc:         "missing result",
+		resultName:   "missing",
+		resultValue:  "",
+		createResult: false,
+		stepError:    false,
+	}} {
+		t.Run(c.desc, func(t *testing.T) {
+			dir := t.TempDir()
+			if c.createResult == true {
+				createResult(dir, c.resultName, c.resultValue, 0644)
+			}
+			dir2 := t.TempDir()
+			createRun(dir2, c.stepError)
+			err := LookForResults(dir2, dir, c.resultName)
+			if err != nil {
+				t.Fatalf("Did not expect any error but got: %v", err)
+			}
+		})
+	}
+}
+
+func createResult(dir string, resultName string, resultValue string, fileMode fs.FileMode) {
+	resultFile := filepath.Join(dir, resultName)
+	err := os.WriteFile(resultFile, []byte(resultValue), fileMode)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
```

Comment:
Pass in a testing.T + use t.Helper() so that any errors are captured in the right test output:

```suggestion
func createResult(t *testing.T, dir string, resultName string, resultValue string) {
	t.Helper()
	resultFile := filepath.Join(dir, resultName)
	err := os.WriteFile(resultFile, []byte(resultValue), 0644)
	if err != nil {
		t.Fatal(err)
	}
}
```

same with createRun.

Reply:
done! Thanks!